### PR TITLE
build: upgrade to node 18

### PR DESCRIPTION
**Pull request by onlyonehas**
upgraded to node 18 
upgraded all dependancies to latest version

### CODE REVIEW

File: templates/spec.hbs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+%define debug_package %{nil}
```
Comment (onlyonehas) [inline templates/spec.hbs:1]: This was added as in centos/rocky8 it throw an error when looking for files which do not exist.
https://forums.centos.org/viewtopic.php?t=74540

File: test/archiver.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const zlib = require('zlib');
 const stream = require('stream');
 const assert = require('assert');
 const sinon = require('sinon');
-const sandbox = sinon.sandbox.create();
+const sandbox = sinon.createSandbox();
```
Comment (onlyonehas) [inline test/archiver.js:9]: creating sandbox in the latest version of sinon is ever so slightly different 
https://sinonjs.org/releases/v15/sandbox/

**Comment by nicholasgriffintn** [comment]
Presumably we'd release this as a major btw if we are to as we're not quite ready for the transition yet :)

**Review by JohnNewman1 — APPROVED** [review]
LGTM

**Pull request merged**